packages/terra-date-picker/src/DateUtil.js

```diff
@@ -2,31 +2,40 @@
 import moment from 'moment';
 
 class DateUtil {
-  // Converts an ISO 8601 date into a moment object. If the date is invalid and unable to convert, the originally provided date is returned.
+  /**
+   * Creates a moment object using the provided date string. Moment is unable to initialize a valid date if the date passed in is
+   * null, empty string, or alpha characters and undefined would be returned.
+   * @param {string|undefined} date - The date to convert. Expect to be in ISO format.
+   * @return {object|undefined} - The moment object. Undefined if unable to convert.
+   */
   static createSafeDate(date) {
-    if (date) {
-      const momentDate = moment(date);
-      return momentDate.isValid() ? momentDate : date;
+    if (!date) {
+      return undefined;
```

Comment:
nit - an empty return statement automatically returns undefined
```suggestion
      return;
```

Reply:
Having an empty return statement here would introduce a [consistent-return](https://eslint.org/docs/rules/consistent-return) linter error since it also needs to return a moment object at the end of this function.